go/mysql/connection.go

```diff
@@ -16,6 +22,7 @@ type ConnectionConfig struct {
 	User       string
 	Password   string
 	ImpliedKey *InstanceKey
+	tlsConfig  *tls.Config
```

Comment:
Any reason not to make this variable public since you have a simple getter down below anyway?

Reply:
Wanted to force callers consuming this struct to set up the TLS for the ConnectionConfig through the `UseTLS` function below if they wanted to set up the tls.Config. This prevents passing around a ConnectionConfig whose tls.Config has not be registered with the mysql driver.